crates/bevy_pbr/src/light_probe/environment_filter.wgsl

```diff
@@ -0,0 +1,396 @@
+    face: u32,
+}
+fn dir_to_cube_uv(dir: vec3f) -> CubeUV {
+    let abs_dir = abs(dir);
+    var face: u32 = 0u;
+    var uv: vec2f = vec2f(0.0);
+    
+    // Find the dominant axis to determine face
+    if (abs_dir.x >= abs_dir.y && abs_dir.x >= abs_dir.z) {
+        // X axis is dominant
+        if (dir.x > 0.0) {
+            face = 0u; // +X
+            uv = vec2f(-dir.z, -dir.y) / dir.x;
+        } else {
+            face = 1u; // -X
+            uv = vec2f(dir.z, -dir.y) / abs_dir.x;
+        }
+    } else if (abs_dir.y >= abs_dir.x && abs_dir.y >= abs_dir.z) {
+        // Y axis is dominant
+        if (dir.y > 0.0) {
+            face = 2u; // +Y
+            uv = vec2f(dir.x, dir.z) / dir.y;
+        } else {
+            face = 3u; // -Y
+            uv = vec2f(dir.x, -dir.z) / abs_dir.y;
+        }
+    } else {
+        // Z axis is dominant
+        if (dir.z > 0.0) {
+            face = 4u; // +Z
+            uv = vec2f(dir.x, -dir.y) / dir.z;
+        } else {
+            face = 5u; // -Z
+            uv = vec2f(-dir.x, -dir.y) / abs_dir.z;
+        }
+    }
+    
+    // Convert from [-1,1] to [0,1]
+    return CubeUV(uv * 0.5 + 0.5, face);
+}
+
+// Sample an environment map with a specific LOD
+fn sample_environment(dir: vec3f, level: f32) -> vec4f {
+    let cube_uv = dir_to_cube_uv(dir);
+    return textureSampleLevel(input_texture, input_sampler, cube_uv.uv, cube_uv.face, level);
+}
+
+// Hammersley sequence for quasi-random points
+fn hammersley_2d(i: u32, n: u32) -> vec2f {
+    let inv_n = 1.0 / f32(n);
+    let vdc = f32(reverseBits(i)) * 2.3283064365386963e-10; // 1/2^32
+    return vec2f(f32(i) * inv_n, vdc);
+}
+
+// Blue noise randomization
+fn sample_noise(pixel_coords: vec2u) -> vec4f {
+    let noise_size = vec2u(u32(constants.blue_noise_size.x), u32(constants.blue_noise_size.y));
+    let noise_coords = pixel_coords % noise_size;
+    let uv = vec2f(noise_coords) / constants.blue_noise_size;
+    return textureSampleLevel(blue_noise_texture, input_sampler, uv, 0.0);
+}
+
+// from bevy_pbr/src/render/pbr_lighting.wgsl
+fn perceptualRoughnessToRoughness(perceptualRoughness: f32) -> f32 {
+    // clamp perceptual roughness to prevent precision problems
+    // According to Filament design 0.089 is recommended for mobile
+    // Filament uses 0.045 for non-mobile
+    let clampedPerceptualRoughness = clamp(perceptualRoughness, 0.089, 1.0);
+    return clampedPerceptualRoughness * clampedPerceptualRoughness;
+}
+
+// GGX/Trowbridge-Reitz normal distribution function (D term)
+// from bevy_pbr/src/render/pbr_lighting.wgsl
+fn D_GGX(roughness: f32, NdotH: f32) -> f32 {
+    let oneMinusNdotHSquared = 1.0 - NdotH * NdotH;
+    let a = NdotH * roughness;
+    let k = roughness / (oneMinusNdotHSquared + a * a);
+    let d = k * k * (1.0 / PI);
+    return d;
+}
+
+// Probability-density function that matches the bounded VNDF sampler (Listing 2)
+fn ggx_vndf_pdf(i: vec3<f32>, NdotH: f32, roughness: f32) -> f32 {
```

Comment:
We should discuss where we want to move this to, since it'll be used for Solari too. Maybe a new ggx_vndf.wgsl file somewhere in bevy_pbr?

Reply:
resolved by https://github.com/mate-h/bevy/pull/8